sklearn/utils/_metadata_requests.py

```diff
@@ -368,13 +381,46 @@ class MetadataRequest:
     def __init__(self, owner):
         # this is used to check if the user has set any request values
         self._is_default_request = False
-        for method in METHODS:
+        self.owner = owner
+        for method in SIMPLE_METHODS:
             setattr(
                 self,
                 method,
                 MethodMetadataRequest(router=self, owner=owner, method=method),
             )
 
+    def __getattr__(self, name):
```

Comment:
Can there be a docstring to explain what this method is doing? At a glance, it looks like it is dynamically creating the metadata routing for the composite methods.

Also, what do you think of setting `COMPOSITE_METHODS` during `MetadataRequest.__init__`? It would increase the size of the object, but we do not need override `__getattr__`.

Reply:
@thomasjpfan  that would require setting them as descriptors (which we can do), and to make it dynamic based on the values in `COMPOSITE_METHODS` it'd need a meta class with `__init_sublcass__` or an `ABC` meta-class. Can surely do, but I'm not sure if it's more understandable/easier than this approach.